ppx/ppx_js/lib_internal/ppx_js_internal.ml

```diff
@@ -563,11 +589,26 @@ let preprocess_literal_object mappper fields :
           | (Immutable | Mutable), [ `Writeonly ] -> `Writeonly
           | (Immutable | Mutable), [ `Readwrite ] -> `Readwrite
           | (Immutable | Mutable), [ `Unkown s ] ->
-              Location.raise_errorf ~loc:exp.pcf_loc "Unkown jsoo attribute ([@@@@%s])." s
+              let err =
+                Location.Error.make
+                  ~loc:exp.pcf_loc
+                  ~sub:[]
+                  (Printf.sprintf "Unkown jsoo attribute ([@@%s])." s)
```

Comment:
Is the error message different now ?

Reply:
The error message is not different. But there is a problem with `Ppxlib.Location.raise_errorf`: using `Ppxlib.Location.raise_errorf` in a rewriter that is tested with toplevel_expect_test causes toplevel_expect_test to raise that error instead of returning a diff. That's why I've rewritten this using `Ppxlib.Location.make` instead of `Ppxlib.Location.raise_errorf`. I haven't found out what is happening to `Ppxlib.Location.raise_errorf`, yet.

Reply:
The issue is that we're catching `Ppxlib.Location.Error` somewhere in that file, but `raise_errorf` (which comes from compiler-libs) raises `Ocaml_common.Location.Error`

Reply:
Thanks! That's good to understand. I'll talk with @NathanReb about the question if we should always raise the `Ppxlib.Location.Error` exception in `Ppxlib.Location`, including in `Ppxlib.Location.raise_errorf`. 

What I've done here now is to simply define `raise_errorf` at the top of the file to keep the diff smaller. Is that good enough?

Reply:
As talked about, I've defined an exception here in order to avoid `Ppxlib.Location.Error` which will probably be removed from the ppxlib API.